query/Fetcher.java

```diff
@@ -331,10 +358,11 @@ public static Projection create(Reasoner reasoner, ConceptManager conceptMgr, Co
                 )) {
                     throw TypeDBException.of(SUBQUERY_UNBOUNDED, key.label());
                 }
-
+                // use a new context to avoid taking limits into sub-query
+                Context.Query subContext = new Context.Query(context.parent(), context.options());
```

Comment:
Bug fix: same as before, we do not pass the outer query's context into the inner one, creating a new context from the transaction context directly.